cmd/clusterctl/cmd/topology.go

```diff
@@ -0,0 +1,306 @@
+		table.Rich(
+			[]string{
+				m.After.GetNamespace(),
+				m.After.GetKind(),
+				m.After.GetName(),
+				"modified",
+			},
+			[]tablewriter.Colors{
+				{}, {}, {}, {tablewriter.FgYellowColor},
+			},
+		)
+	}
+
+	// Add the deleted rows.
+	sort.Slice(out.Deleted, func(i, j int) bool {
+		return out.Deleted[i].GetKind() < out.Deleted[j].GetKind()
+	})
+	for _, d := range out.Deleted {
+		table.Rich(
+			[]string{
+				d.GetNamespace(),
+				d.GetKind(),
+				d.GetName(),
+				"deleted",
+			},
+			[]tablewriter.Colors{
+				{}, {}, {}, {tablewriter.FgRedColor},
+			},
+		)
+	}
+	table.Render()
+}
+
+func writeOutputFiles(out *cluster.DryRunOutput, outDir string) error {
+	if _, err := os.Stat(outDir); os.IsNotExist(err) {
+		fmt.Printf("The output directory %q could not be detected.\n", outDir)
+		return fmt.Errorf("output directory does not exist")
+	}
+
+	// Write created files
+	createdDir := path.Join(outDir, "created")
+	if _, err := os.Stat(createdDir); err != nil && os.IsNotExist(err) {
+		if err := os.Mkdir(createdDir, 0750); err != nil {
+			return errors.Wrapf(err, "failed to create %q directory", createdDir)
+		}
+	}
+	for _, c := range out.Created {
+		yaml, err := utilyaml.FromUnstructured([]unstructured.Unstructured{*c})
+		if err != nil {
+			return errors.Wrap(err, "failed to convert object to yaml")
+		}
+		fileName := fmt.Sprintf("%s_%s_%s.yaml", c.GetKind(), c.GetNamespace(), c.GetName())
+		filePath := path.Join(createdDir, fileName)
+		f, err := os.OpenFile(filePath, os.O_CREATE|os.O_TRUNC|os.O_RDWR, 0600) //nolint:gosec
+		if err != nil {
+			return errors.Wrapf(err, "failed to created file %q", filePath)
+		}
+		if _, err := f.Write(yaml); err != nil {
+			return errors.Wrapf(err, "failed to write yaml to file %q", filePath)
+		}
+	}
+	if len(out.Created) != 0 {
+		fmt.Printf("Created objects are written to directory %q\n", createdDir)
+	}
+
+	// Write modified files
+	modifiedDir := path.Join(outDir, "modified")
+	if _, err := os.Stat(modifiedDir); err != nil && os.IsNotExist(err) {
+		if err := os.Mkdir(modifiedDir, 0750); err != nil {
+			return errors.Wrapf(err, "failed to create %q directory", modifiedDir)
+		}
+	}
+	for _, m := range out.Modified {
+		// Write the modified object to file.
+		yamlModified, err := utilyaml.FromUnstructured([]unstructured.Unstructured{*m.After})
+		if err != nil {
+			return errors.Wrap(err, "failed to convert object to yaml")
+		}
+		fileNameModified := fmt.Sprintf("%s_%s_%s.modified.yaml", m.After.GetKind(), m.After.GetNamespace(), m.After.GetName())
+		filePathModified := path.Join(modifiedDir, fileNameModified)
+		fModified, err := os.OpenFile(filePathModified, os.O_CREATE|os.O_TRUNC|os.O_RDWR, 0600) //nolint:gosec
+		if err != nil {
+			return errors.Wrapf(err, "failed to created file %q", filePathModified)
+		}
+		if _, err := fModified.Write(yamlModified); err != nil {
+			return errors.Wrapf(err, "failed to write yaml to file %q", filePathModified)
+		}
```

Comment:
Could it be that this block is 3x the same? If yes could we split it out in a separate func please?